src/mysql.c

```diff
@@ -1014,6 +1091,8 @@ static int mysql_read (user_data_t *ud)
 
 		gauge_submit ("cache_size", "qcache",
 				qcache_queries_in_cache, db);
+ 		gauge_submit ("cache_queries", "qcache-free-memory",
```

Comment:
I don't think what strong recommendation of using existing types should apply.
What other type instances could be added to this type? 
That type instance _MUST_ be related to cache, so we will unable to use this _type_ to other things. 
The 'memory' type is too general for this. 

In my view, this metric may be sent as 'cache_size', but this already used for 'queries number', so new type should appear. In this case, I recommend to send this as 'cache_memory' with type instance 'free'. 

So, when new metric will appear in 'show global status' output we will add it with no problems.